compiler/src/dmd/dsymbolsem.d

```diff
@@ -6812,6 +6830,140 @@ private void aliasAssignSemantic(AliasAssign ds, Scope* sc)
     ds.semanticRun = PASS.semanticdone;
 }
 
+/***************************************
+ * Expands template instance arguments inside 'alias assign' target declaration (aliassym),
+ * instead of inside 'tempinst.tiargs' every time.
+ * Params:
+ *      tempinst = AliasSeq instance
+ *      aliassym = the AliasDeclaration corresponding to AliasAssign
+ * Returns:
+ *       null.
+ */
+private TupleDeclaration aliasAssignInPlace(Scope* sc, TemplateInstance tempinst,
+                                            AliasDeclaration aliassym)
+{
+    // Mark instance with semantic done, not needed but just in case.
+    tempinst.inst = tempinst;
+    tempinst.semanticRun = PASS.semanticdone;
+    auto tempdecl = tempinst.tempdecl.isTemplateDeclaration();
+    auto ttpident = (*tempdecl.parameters)[0].isTemplateTupleParameter().ident;
+    TupleDeclaration td;
+    if (aliassym.type)
+    {
+        // Convert TypeTuple to TupleDeclaration to avoid back and forth allocations
+        // in the assignment process
+        if (auto tt = aliassym.type.isTypeTuple())
+        {
+            auto objs = new Objects(tt.arguments.length);
+            foreach (i, p; *tt.arguments)
+                (*objs)[i] = p.type;
+            td = new TupleDeclaration(tempinst.loc, ttpident, objs);
+            td.storage_class |= STC.templateparameter;
+            td.building = true;
+            aliassym.type = null;
+        }
+        else if (aliassym.type.isTypeError())
+            return null;
+
+    }
+    else if (auto otd = aliassym.aliassym.isTupleDeclaration())
+    {
+        if (otd.building)
+            td = otd;
+        else
+        {
+            td = new TupleDeclaration(tempinst.loc, ttpident, otd.objects.copy());
+            td.building = true;
+            td.storage_class |= STC.templateparameter;
+        }
+    }
+    // If starting from single element in aliassym (td == null) we need to build the tuple
+    // after semanticTiargs to keep same semantics (for example a FuncLiteraldeclaration
+    // template argument is converted to FuncExp)
+    if (td)
+        aliassym.aliassym = td;
+    aliassym.semanticRun = PASS.semanticdone;
+    if (!TemplateInstance.semanticTiargs(tempinst.loc, sc, tempinst.tiargs, 0, td))
+    {
+        tempinst.errors = true;
+        return null;
+    }
+    // The alias will stop tuple 'building' mode when used (in AliasDeclaration.toAlias(),
+    // then TupleDeclaration.getType() will work again)
+    aliassym.semanticRun = PASS.initial;
+
+    if (!td)
+        td = new TupleDeclaration(tempinst.loc, ttpident, tempinst.tiargs);
+    else
+    {
+        auto tiargs = tempinst.tiargs;
+        size_t oldlen = td.objects.length;
+        size_t origstart;
+        size_t insertidx;
+        size_t insertlen;
+        size_t i;
+        foreach (o; *tiargs)
+        {
+            if (o !is td)
+                ++insertlen;
```

Comment:
if (o !is td)
{
    ++insertlen;
    ++i;
    continue;
}

Reply:
Done. Integrated `i` as `foreach` variable too.